cas-server-webapp-actions/src/main/java/org/jasig/cas/web/flow/InitialFlowSetupAction.java

```diff
@@ -67,13 +65,23 @@
     @Override
     protected Event doExecute(final RequestContext context) throws Exception {
         final HttpServletRequest request = WebUtils.getHttpServletRequest(context);
-        if (!this.pathPopulated) {
-            final String contextPath = context.getExternalContext().getContextPath();
-            final String cookiePath = StringUtils.hasText(contextPath) ? contextPath + '/' : "/";
-            logger.info("Setting path for cookies to: {} ", cookiePath);
+
+        final String contextPath = context.getExternalContext().getContextPath();
+        final String cookiePath = StringUtils.isNotBlank(contextPath) ? contextPath + '/' : "/";
+
+        if (StringUtils.isBlank(warnCookieGenerator.getCookiePath())) {
```

Comment:
Note that this patch provides now an extra step for the deployer because previously deployers simply had to deploy CAS where they wanted to, and the cookie would have been automatically set. This now requires them explicitly configure the cookie as well, but this is more of a special case since almost all deployments opt for /cas.